src/fallback.rs

```diff
@@ -364,7 +364,10 @@ impl FileInfo {
 
     fn source_text(&self, span: Span) -> String {
         let lo = (span.lo - self.span.lo) as usize;
-        let trunc_lo = &self.source_text[lo..];
+        let trunc_lo = match self.source_text.char_indices().nth(lo) {
```

Comment:
This makes it _O(n²)_ to compute the source_text of lots of tokens in the same stream, which is probably an important use case. I'll follow up with a separate PR to amortize this.

```rust
for token in token_stream {
    let _ = token.span().source_text();
}
```

Reply:
#412